Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -0,0 +1,1218 @@
+            GPRMHist.SetFilter(DOCDATE, '>=%1', System.DMY2Date(1, 1, InitialHistYear));
+
+        if not GPRMHist.FindSet() then
+            exit;
+
+        repeat
+            Clear(HistReceivablesDocument);
+
+            if GPRM00101.Get(GPRMHist.CUSTNMBR) then
+                CustomerName := GPRM00101.CUSTNAME;
+
+#pragma warning disable AA0139
+            HistReceivablesDocument."Document No." := GPRMHist.DOCNUMBR.TrimEnd();
+            HistReceivablesDocument."Customer No." := GPRMHist.CUSTNMBR.TrimEnd();
+#pragma warning restore AA0139
+
+            HistReceivablesDocument."Customer Name" := CustomerName;
+            HistReceivablesDocument."Document Type" := ConvertGPDocTypeToHistReceivablesDocType(GPRMHist.RMDTYPAL);
+            HistReceivablesDocument."Batch No." := GPRMHist.BACHNUMB;
+            HistReceivablesDocument."Batch Source" := GPRMHist.BCHSOURC;
+            HistReceivablesDocument."Audit Code" := GPRMHist.TRXSORCE;
+            HistReceivablesDocument."Trx. Description" := GPRMHist.TRXDSCRN;
+            HistReceivablesDocument."Document Date" := GPRMHist.DOCDATE;
+            HistReceivablesDocument."Due Date" := GPRMHist.DUEDATE;
+            HistReceivablesDocument."Post Date" := GPRMHist.POSTDATE;
+            HistReceivablesDocument."User" := GPRMHist.PSTUSRID;
+            HistReceivablesDocument."Currency Code" := CopyStr(GPRMHist.CURNCYID, 1, MaxStrLen(HistReceivablesDocument."Currency Code"));
+            HistReceivablesDocument."Orig. Trx. Amount" := GPRMHist.ORTRXAMT;
+            HistReceivablesDocument."Current Trx. Amount" := GPRMHist.CURTRXAM;
+            HistReceivablesDocument."Sales Amount" := GPRMHist.SLSAMNT;
+            HistReceivablesDocument."Cost Amount" := GPRMHist.COSTAMNT;
+            HistReceivablesDocument."Freight Amount" := GPRMHist.FRTAMNT;
+            HistReceivablesDocument."Misc. Amount" := GPRMHist.MISCAMNT;
+            HistReceivablesDocument."Tax Amount" := GPRMHist.TAXAMNT;
+            HistReceivablesDocument."Disc. Taken Amount" := GPRMHist.DISTKNAM;
+            HistReceivablesDocument."Customer Purchase No." := GPRMHist.CSPORNBR;
+            HistReceivablesDocument."Salesperson No." := GPRMHist.SLPRSNID;
+            HistReceivablesDocument."Sales Territory" := GPRMHist.SLSTERCD;
+            HistReceivablesDocument."Ship Method" := GPRMHist.SHIPMTHD;
+            HistReceivablesDocument."Cash Amount" := GPRMHist.CASHAMNT;
+            HistReceivablesDocument."Commission Dollar Amount" := GPRMHist.COMDLRAM;
+            HistReceivablesDocument."Invoice Paid Off Date" := GPRMHist.DINVPDOF;
+            HistReceivablesDocument."Payment Terms ID" := GPRMHist.PYMTRMID;
+            HistReceivablesDocument."Write Off Amount" := GPRMHist.WROFAMNT;
+
+            if not HistReceivablesDocument.Insert() then
+                HistMigrationStatusMgmt.ReportLastError("Hist. Migration Step Type"::"GP Receivables Trx.", GPRMHist.DOCNUMBR, true);
```

Comment:
We may need the code to fix errors somehow or to try moving failed records again.